tests/BaseTest.sol

```diff
@@ -42,6 +45,135 @@ contract BaseTest is Test {
     uint256 envelopeNonce;
   }
 
+  modifier generateEmergencyState(address ccc) {
+    address clEmergencyOracle = IEmergencyConsumer(ccc).getChainlinkEmergencyOracle();
+    vm.mockCall(
+      clEmergencyOracle,
+      abi.encodeWithSelector(ICLEmergencyOracle.latestRoundData.selector),
+      abi.encode(uint80(2), int256(1), block.timestamp, block.timestamp, uint80(2))
+    );
+    _;
+  }
+
+  modifier validateEmergencySolved(address ccc) {
+    _;
+    address clEmergencyOracle = IEmergencyConsumer(ccc).getChainlinkEmergencyOracle();
+    (, int256 answer, , , ) = ICLEmergencyOracle(clEmergencyOracle).latestRoundData();
+
+    uint256 emergencyCount = IEmergencyConsumer(ccc).getEmergencyCount();
+
+    assertEq(emergencyCount, uint256(answer));
+  }
+
+  modifier generateRetryTxState(
+    address cccOwner,
+    address ccc,
+    uint256 destinationChainId,
+    address destination,
+    uint256 gasLimit
+  ) {
+    vm.assume(gasLimit < 300_000);
+    // set caller as an approved sender
+    address[] memory senders = new address[](1);
+    senders[0] = address(this);
+
+    vm.startPrank(cccOwner);
+    ICrossChainForwarder(ccc).approveSenders(senders);
+    vm.stopPrank();
+
+    // call forward message
+    ICrossChainForwarder(ccc).forwardMessage(destinationChainId, destination, gasLimit, MESSAGE);
+    _;
+  }
+
+  function deployCCC(
```

Comment:
Why do we neen CCC deployed?
If you want integration test, better to do it against real CCC on some network. But not sure that it's should be here